src/server/game/Entities/Creature/TemporarySummon.cpp

```diff
@@ -300,6 +300,21 @@ void Minion::RemoveFromWorld()
     TempSummon::RemoveFromWorld();
 }
 
+void Minion::setDeathState(DeathState s)
+{
+    Creature::setDeathState(s);
+    if (s == JUST_DIED && IsGuardianPet())
```

Comment:
That just looks so horrible.. couldn't you please do it like:

if (s != JUST_DIED || !IsGuardianPet())
    return;

and so on.. that would look a lot better :D

Reply:
Smth like that

`void Minion::setDeathState(DeathState s)
{
    Creature::setDeathState(s);
    if (s != JUST_DIED || !IsGuardianPet())
        return;

    Unit* owner = GetOwner();
    if (!owner || !owner->IsPlayer() || owner->GetMinionGUID() != GetGUID())
        return;

    for (Unit* controlled : owner->m_Controlled)
    {
        if (controlled->GetEntry() == GetEntry() && controlled->IsAlive())
        {
            owner->SetMinionGUID(controlled->GetGUID());
            owner->SetPetGUID(controlled->GetGUID());
            owner->ToPlayer()->CharmSpellInitialize();
        }
    }
}
`

Reply:
Fixed. Thanks!